### PR TITLE
Adding capability to set serviceAccountName through flags

### LINKED ISSUE #83: As an app dev or app opp, I can CRUD the workload.spec.serviceAccountName via command flag in `create/update/apply`
### Description of problem
As of #76, it's possible to CRUD the workload `spec.serviceAccountName` via `workload.yaml`
As the community engages more deeply with TCE's App Toolkit, the desire/need to specify a non-default serviceAccountName for individual workloads is likely to increase over time.
Providing the `--serviceAccountName` flag will facilitate doing so for those who don't have the skill or desire to engage with yaml.

### Proposed solution

**new service-account flag**
```gherkin
Given i have a workload to CRUD
When i provide "--service-account foo" in my "workload create/update/apply"
Then my submitted workload object will include "spec.serviceAccountName: foo" property/value
```

**new service-account flag alias**
```gherkin
Given i have a workload to CRUD
When i provide "--sa foo" in my "workload create/update/apply"
Then my submitted workload object will include "spec.serviceAccountName: foo" property/value
```

**delete service-account approach**
```gherkin
Given i have a workload which has a `spec.serviceAccountName: foo` to CRUD
When i provide "--service-account '' (single or double quotes should be acceptable)
 " in my "workload create/update/apply"
Then my submitted workload object will delete the "spec.serviceAccountName: foo" property/value
```

### Example

<Code snippets that illustrate the when/then blocks>

### Describe alternatives you've considered
Providing support only via `-f workload.yaml`

cc @kartiklunkad26

### PR DESCRIPTION
**Pull request by warango4**
Pull request

### What this PR does / why we need it
In this feature, serviceAccountName is added as a flag and can be used in create/update/apply commands.
To delete serviceAccountName from workload, the flag should be set to "" or ''

E.g.
- `tanzu apps workload create rmq-sample-app --git-repo https://github.com/jhvhs/rabbitmq-sample --git-branch main --type web --service-account myServiceAccount -y`
- `tanzu apps workload apply rmq-sample-app --service-account myServiceAccountUpdated -y // When retrieving the workload, the new service account name should be 'myServiceAccountUpdated'`
- `tanzu apps workload update rmq-sample-app --service-account "" -y // Should delete serviceAccountName from workload`

### Which issue(s) this PR fixes

Fixes #83 

### Describe testing done for PR

- Added unit test
- Tested in local cluster

### Additional information or special notes for your reviewer

Signed-off by: Wendy Arango warango@vmware.com

### CODE REVIEW

**Comment by warango4** [comment]
What this PR is missing 
- Shorthand for `--service-account` as mentioned in issue (`--sa`)
- Usage string for new flag (missing design)

File: pkg/flags/flags.go [hunk 1]
```diff
@@ -21,47 +21,48 @@ import (
...
+	LimitMemoryFlagName        = "--limit-memory"
+	LiveUpdateFlagName         = "--live-update"
+	LocalPathFlagName          = "--local-path"
+	NamespaceFlagName          = cli.NamespaceFlagName
+	NoColorFlagName            = cli.NoColorFlagName
+	OutputFlagName             = "--output"
+	ParamFlagName              = "--param"
+	RequestCPUFlagName         = "--request-cpu"
+	RequestMemoryFlagName      = "--request-memory"
+	ServiceAccountNameFlagName = "--service-account"
```
Comment (shashwathi) [inline pkg/flags/flags.go:55]: nit: this const can be renamed `ServiceAccountFlagName`

Comment (shashwathi) [inline pkg/flags/flags.go:55]: nit: this const can be renamed `ServiceAccountFlagName`

File: pkg/apis/cartographer/v1alpha1/workload_helpers.go [hunk 2]
```diff
@@ -78,8 +78,12 @@ func (w *Workload) loadAndValidateDocuments(in io.Reader) error {
 	return nil
 }
 
-func (w *Workload) MergeServiceAccountName(serviceAccountName string) {
-	w.Spec.ServiceAccountName = serviceAccountName
+func (w *WorkloadSpec) MergeServiceAccountName(serviceAccountName string) {
```
Comment (shashwathi) [inline pkg/apis/cartographer/v1alpha1/workload_helpers.go:81]: Add unit tests for this func

**Review by rashedkvm — COMMENTED** [review]
> * Usage string for new flag (missing design)

@warango4 please update PR with design input

**Review by rashedkvm — COMMENTED** [review]
How does the user know to use an empty string so an existing `serviceAccountName` value can be removed?

**Review by shashwathi — APPROVED** [review]

**Review by rashedkvm — APPROVED** [review]

**Pull request merged**